### PR TITLE
FirstPay rest support for ApplePay GooglePay

### PR DESCRIPTION
**Pull request by sinourain**
Description
-------------------------
Spreedly reference:
[SER-1099](https://spreedly.atlassian.net/browse/SER-1099)
[SER-1100](https://spreedly.atlassian.net/browse/SER-1100)

This commit contains:
- Add Google Pay
- Add Apple Pay
- Add unit and remote tests

Unit test
-------------------------
Finished in 26.44884 seconds.
5826 tests, 79142 assertions, 2 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications 99.9657% passed
220.27 tests/s, 2992.27 assertions/s

Remote test
-------------------------
Finished in 26.44884 seconds.
5826 tests, 79142 assertions, 2 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications 99.9657% passed
220.27 tests/s, 2992.27 assertions/s

Rubocop
-------------------------
792 files inspected, no offenses detected

### CODE REVIEW

File: test/unit/gateways/first_pay_json_test.rb [hunk 1]
```diff
@@ -55,6 +75,58 @@ def test_failed_purchase
...
+      assert_match(/cardExpYear/, data)
+      assert_match(/otherFields/, data)
+      assert_match(/ownerName/, data)
+      assert_match(/ownerStreet/, data)
+      assert_match(/ownerCity/, data)
+      assert_match(/ownerState/, data)
+      assert_match(/ownerZip/, data)
+      assert_match(/ownerCountry/, data)
+      assert_match(/processorId/, data)
+      assert_match(/merchantKey/, data)
```
Comment (gasb150) [inline test/unit/gateways/first_pay_json_test.rb:120]: 💭 Opinion:

Could be a good idea to check if the `:walletType` is `apple_pay` or `google_pay` (according to the case) and check if the request contains the `paymentCryptogram` and `eciIndicator`.

**Review by gasb150 — COMMENTED** [review]
LGTM, just a minor comment.

File: test/unit/gateways/first_pay_json_test.rb (hunk 1 above)
Comment (sinourain) [reply test/unit/gateways/first_pay_json_test.rb]: Done!

**Review by naashton — CHANGES_REQUESTED** [review]
@sinourain - could you fix the merge conflicts and re-tag me for review, just so it is easier for me to check the changes specific to this PR.

**Comment by sinourain** [comment]
> @sinourain - could you fix the merge conflicts and re-tag me for review, just so it is easier for me to check the changes specific to this PR.

Done!

**Review by naashton — APPROVED** [review]
Nice work!

File: lib/active_merchant/billing/gateways/first_pay/first_pay_json.rb [hunk 2]
```diff
@@ -75,6 +81,7 @@ def scrub(transcript)
           gsub(%r(("processorId\\?"\s*:\s*\\?")[^"]*)i, '\1[FILTERED]').
           gsub(%r(("merchantKey\\?"\s*:\s*\\?")[^"]*)i, '\1[FILTERED]').
           gsub(%r(("cardNumber\\?"\s*:\s*\\?")[^"]*)i, '\1[FILTERED]').
+          gsub(%r(("paymentCryptogram\\?"\s*:\s*\\?")[^"]*)i, '\1[FILTERED]').
```
Comment (naashton) [inline lib/active_merchant/billing/gateways/first_pay/first_pay_json.rb:84]: Can we include a test to validate that we are scrubbing this value correctly?

**Review by naashton — CHANGES_REQUESTED** [review]
Looks pretty good. I just have one request to see a test that validates we are scrubbing the cryptogram correctly. And also if you could include a changelog entry with your update :D Thanks, @sinourain!

File: lib/active_merchant/billing/gateways/first_pay/first_pay_json.rb (hunk 2 above)
Comment (sinourain) [reply lib/active_merchant/billing/gateways/first_pay/first_pay_json.rb:84]: Done!

**Review by naashton — APPROVED** [review]
LGTM! Thanks, Luis!

**Pull request merged**